### DISCUSSION TITLE
Version 0.38 is out

### DISCUSSION THREAD

**Discussion by migueldeicaza** [migueldeicaza/SwiftGodot#393] [Announcements]
This is a juicy one, enjoy!

Documentation has been updated accordingly:

* Many new framework type tests by Mikhail Tishin and many defaults in
  our core types to match the default values in Godot.  Thank you
  Mikhail!

* Methods for built-in types now also follow the pattern of the object
  types of Godot, if the first parameter name matches the last word of
  the method, the name is dropped (Mikhail Tishin)

* Additional diagnostics in the use of macros, it will now flag when a
  @Callable is applied to an async or throwing function (Jae Kong)

* Bug fix #342: Fix editor crash when trying to set mismatched value
  to exported property.  Mikhail Tishin

* Workaroudn a Swift-ism that passes UnsafeRawPointers representing
  nil, but not being a UnsafeRawPointer - helps catch bugs when we get
  a null back from godot (Miguel)

* Fixes Signal dispatching, these were incorrectly encoded and would
  crash your application (Miguel)

* Fixes #exportGroup was applying the prefix to all subsequent
  exported properties' propertyName - Estevan Hernandez

* Adds support for #exportSubgroups - Estevan Hernandez

* Adds support for keyed indexer for built-in Types (Mikhail Tishin),
  fixes bugs #45 and #46

* Prevent SwiftGodot subclasses from executing in editor by default
  (Mikhail Tishin).  This brings a Godot @tool-like setup, you can
  control this by passing calling the @Godot macro with either (.tool)
  or (.gameplay).

* Fixes to the Snapping methods (Mikhail Tishin)

* New unary minus operatos for Vectors (Mikhail Tishin)

* Fixes for Hot Reloading (Mikhail Tishin), but we still have an issue
  with Godot or Swift not fully unloading libraries that is being
  tracked (https://github.com/migueldeicaza/SwiftGodot/issues/384

* Fixes exporting of fields named `value` Airyzz.

* Add support for calling variant methods and access elements of
  arrays held in variants (Miguel)

* Makes the `NodePath` description method match the Godot method that
  produces their name - as Godot does not surface this method (Miguel)

* New convenience for PackedByteArray: `asData()` returns the byte
  array as a Swift Data type.

* Color now exposes HSV properties (Mikhail Tishin)

* Variant memory leak fixes.  Miguel See the discussion here for
  additional information:
  https://github.com/migueldeicaza/SwiftGodot/discussions/388

* StringName and GString are now Hashable (Mikhail Tishin)

**Comment by VojtaStruhar** [comment]
Super excited about this project! Keep it up! ✨
